docs/dev_guide/maintainer_workflow.rst

```diff
@@ -0,0 +1,143 @@
+.. _maintainer-workflow:
+
+Workflow for Maintainers
+========================
+
+This page is for maintainers |emdash| those of us who merge our own or other
+peoples' changes into the upstream repository.
+
+Being as how you're a maintainer, you are completely on top of the basic stuff
+in :ref:`version_control`.
+
+Integrating changes
+-------------------
+
+Whenever possible, merge pull requests automatically via the pull request manager on GitHub. Merging should only be done manually if there is a really good reason to do this!
+
+Please try to make sure that pull requests do not contain a messy history with merges, etc. If this is the case, then you have two options. To use the rebase merge on Github or to follow the manual instructions, and make sure the fork is rebased to tidy the history before committing. The second option should only be used on complex pull requests.
+
+
+Using Milestones and Labels
+===========================
+
+These guidelines are adapted from `similar guidelines <http://docs.astropy.org/en/stable/development/workflow/maintainer_workflow.html#using-milestones-and-labels>`_
+.. _milestones-and-labels:
+followed by AstroPy:
+
+* 100% of confirmed issues and new features should have a milestone
```

Comment:
Actually it's really good that you digged this up now, and we had a few changes that is policy but is not really updated here. e.g. we don't milestone issues, unless they are release critical or for some other reasons should/aimed to be addressed for a release. This is to avoid the ugliness of constant remilestoning issues (and in practice using a meaningless "future" or similar milestone is not very helpful either).
But every PR has to have a milestone.

Reply:
That makes a lot of sense, I am in favor of adopting that.

Reply:
So you not use a future milestone in practice?

Reply:
There are issues still milestoned with it, but I support it to be removed, https://github.com/astropy/astropy/issues/5658#issuecomment-360499744. Since you start fresh, you can avoid this mistake :)

Reply:
Yeah, we might just do that. I guess it would be similar to @Cadair's idea of a Finish SunPy issue.

Reply:
I'm not sure what "Finish SunPy" is, but suspect I would label it with "Invalid", or treat it as one of those April 1st ones ;)